src/include/simeng/OS/SimOS.hh

```diff
@@ -108,6 +108,25 @@ class SimOS {
   std::unique_ptr<HostBackedFileMMaps> const hfmmap_ =
       std::make_unique<HostBackedFileMMaps>();
 
+  /** Retrieve the simulated nanoseconds elapsed since the core started. */
+  uint64_t getSystemTimer() const;
+
+  /** Receive the syscall from a core and pass onto the syscall handler. */
+  void receiveSyscall(SyscallInfo syscallInfo) const;
+
+  /** Send the syscall result generated by the SyscallHandler to the relevant
+   * core. */
+  void sendSyscallResult(const SyscallResult result) const;
+
+  /** Function to expose the mapfd function within the HostBackedFileMMaps
+   * `hfmmap_` class. */
+  HostFileMMap mmapHostFd(int fd, size_t len, off_t offset);
```

Comment:
Given that `hfmmap` is public is this function needed?

Reply:
Removed